### PR TITLE
Implement network interfaces logging at exit

### PR DESCRIPTION
**Pull request by taiki45**
Signed-off-by: Taiki Ono <taiki@tetrate.io>

Fixes https://github.com/tetratelabs/getenvoy/issues/40

`Interfaces()` is just a wrapper of golang std's `Interfaces()` and it safely picks up interfaces. https://github.com/shirou/gopsutil/blob/v2.19.8/net/net.go#L212 And we can use JSON format to log interfaces at that time safely https://github.com/shirou/gopsutil/blob/v2.19.8/net/net.go#L193.

Output example:

```
[{"mtu":65536,"name":"lo","hardwareaddr":"","flags":["up","loopback"],"addrs":[{"addr":"127.0.0.1/8"},{"addr":"::1/128"}]},{"mtu":1500,"name":"eno1","hardwareaddr":"","flags":["up","broadcast","multicast"],"addrs":[{"addr":"192.168.11.7/24"},{"addr":""},{"addr":""},{"addr":""}]},{"mtu":1500,"name":"docker0","hardwareaddr":"","flags":["up","broadcast","multicast"],"addrs":[{"addr":"172.17.0.1/16"}]}]
```

### CODE REVIEW

File: pkg/binary/envoy/debug/node.go [hunk 1]
```diff
@@ -118,3 +120,19 @@ func safeProc(p *process.Process) *proc {
 		cmd:      cmd,
 	}
 }
+
+func networkInterfaces(r binary.Runner) error {
+	f, err := os.Create(filepath.Join(r.DebugStore(), "node/network_interface.txt"))
```
Comment (liamawhite) [inline pkg/binary/envoy/debug/node.go]: If the file is json can we make this `.json`?

File: pkg/binary/envoy/debug/node_test.go [hunk 2]
```diff
@@ -41,3 +41,21 @@ func Test_ps(t *testing.T) {
 		}
 	})
 }
+
+func Test_network_inerfaces(t *testing.T) {
```
Comment (liamawhite) [inline pkg/binary/envoy/debug/node_test.go]: This test should be integrated with the one above as they do the same things. Add a table-driven test for each file and use only a single run.

Comment (taiki45) [reply pkg/binary/envoy/debug/node_test.go]: Right 👍

**Comment by taiki45** [comment]
@liamawhite Good catch, updated the PR description.

**Comment by liamawhite** [comment]
@taiki45 FYI, that output example isn't valid JSON. The array elements are missing trailing commas.

**Comment by taiki45** [comment]
@liamawhite ah, I filtered hardware addresses and v6 addresses, so it's just my mistake. updated.

**Pull request merged**